auction.sympl

```diff
@@ -33,95 +40,324 @@ schema Auction:
+    else:
+        cvm.error(f"Only the current admin may perform this function.")
+
+@executable
+def _create_premium_channel(premium_channel: ChannelName) -> None:
+    if check_admin():
+        existing_premium = cvm.storage.get(PUBLIC, PremiumChannelStatic, Identifier('premium'))
+        if isinstance(existing_premium, None):
+            premium = PremiumChannel(channel_id=premium_channel, members=[cvm.tx.key_alias])
+            cvm.storage.put(Identifier('premium'), premium)
+            cvm.create_event(f'Premium Channel {premium_channel} Created', std.json(premium))
+        else:
+            cvm.error(f"Premium auction channel already created.")
+    else:
+        cvm.error(f"Only the current admin may perform this function.")
+
+
+#############################################
+# Add/Remove Member API and Implementations #
+#############################################
+
+# Adds a member to the premium channel, giving them access to create and bid on auctions there.
+@clientside
+def add_member(new_member: KeyAlias) -> None:
+    if check_admin():
+        premium = cvm.storage.get(PUBLIC, PremiumChannelStatic, Identifier('premium'))
+        if isinstance(premium, PremiumChannel):
+            with PostTxArgs(PUBLIC):
+                _add_member(new_member)
+        else:
+            cvm.error(f"Channel {premium} is not the Premium channel.")
+    else:
+        cvm.error(f"Only the current admin may perform this function.")
+    
+@executable
+def _add_member(new_member : KeyAlias) -> None:
+    if check_admin():
+        premium = cvm.storage.get(PUBLIC, PremiumChannelStatic, Identifier('premium'))    
+        if isinstance(premium, PremiumChannel):
+            nm_ka : str = new_member
+            for member in premium.members:
+                m_ka : str = member
+                if nm_ka == m_ka:
+                    cvm.error(f"{new_member} is already a member of the premium channel.")
+                    
+            cvm.send_key(premium.channel_id, new_member)
+            premium.members += [new_member]
+            cvm.storage.put(Identifier('premium'), premium)
+            cvm.create_event('Member Added to Premium', std.json(premium))
+
+        else:
+            cvm.error(f"Channel {premium} is not the Premium channel.")
+    else:
+        cvm.error(f"Only the current admin may perform this function.")
+
+# Removes a member from the premium channel.
+@clientside
+def remove_member(old_member: KeyAlias) -> None:
+    if check_admin():
+        premium = cvm.storage.get(PUBLIC, PremiumChannelStatic, Identifier('premium'))
+        if isinstance(premium, PremiumChannel):
+            with PostTxArgs(PUBLIC):
+                _remove_member(old_member)
+        else:
+            cvm.error(f"Channel {premium} is not the Premium channel.")
+    else:
+        cvm.error(f"Only the current admin may perform this function.")
+
+@executable
+def _remove_member(old_member: KeyAlias) -> None:
+    if check_admin():
```

Comment:
There are quirks around if statements with this lang

```
if not check_admin:
    cvm.error("Not admin")
```

I believe is more performant